docker-images/kafka/scripts/zookeeper_run.sh

```diff
@@ -51,5 +53,34 @@ fi
 
 . ./set_kafka_gc_options.sh
 
+############ ZK Upgrade Start ############
+# ZK 3.4.x -> 3.5.x upgrade adds snapshot checks that can cause ZK pods with persistent storage to fail on startup
+# The section below disables the snapshot checks until the servers have created a snapshot.
+
+# If this is a 3.4.x server then place an indicator file in the data directory so this persists over multiple restarts
+flag_file=${ZOOKEEPER_DATA_DIR}/zk_upgrade_from_3_4
+if [[ $(ls libs | grep -Po 'zookeeper-\K3.4.\d+' | head -1) ]]; then
```

Comment:
So I did originally do that, but I went with a containers side only approach so that there is less to take out of the CO when we no longer need to do this in 2 releases time.

Reply:
Jakub has pointed out that the script-only approach will not work for 0.15 ZK 3.4.x -> 0.16 ZK 3.5.6 upgrades so I will switch it back to using a CO set env var to signal the container to set the flag.